sql/analyzer/optimization_rules.go

```diff
@@ -282,14 +282,20 @@ func simplifyFilters(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope,
 		return node, transform.SameTree, nil
 	}
 
-	return transform.Node(node, func(node sql.Node) (sql.Node, transform.TreeIdentity, error) {
+	return transform.NodeWithOpaque(node, func(node sql.Node) (sql.Node, transform.TreeIdentity, error) {
 		filter, ok := node.(*plan.Filter)
 		if !ok {
 			return node, transform.SameTree, nil
 		}
 
 		e, same, err := transform.Expr(filter.Expression, func(e sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
 			switch e := e.(type) {
+			case *plan.Subquery:
```

Comment:
Need an engine test case for this change, no?

Reply:
So this does apply to a bunch of enginetests, but it's only necessary for the new name resolution. It's just a duplicate/idempotent for the regular enginetests tests.